plugin/src/main/java/org/jenkinsci/plugins/structs/SymbolLookup.java

```diff
@@ -68,11 +70,55 @@
         }
     }
 
+    /**
+     * Looks for a {@link Descriptor} that has the given symbol
+     *
+     * @param type
+     *      Restrict the search to a subset of {@link Describable}
+     */
+    public <T> Descriptor<? extends T> findDescriptor(Class<T> type, String symbol) {
+        try {
+            Key k = new Key("findDescriptor",type,symbol);
+            Object i = cache.get(k);
+            if (i!=null)    return (Descriptor)i;
+
+            // not allowing @Symbol to use an invalid identifier.
+            // TODO: compile time check
+            if (!Utilities.isJavaIdentifier(symbol))
+                return null;
+
+            for (Class<?> e : Index.list(Symbol.class, pluginManager.uberClassLoader, Class.class)) {
+                if (Descriptor.class.isAssignableFrom(e)) {
+                    Symbol s = e.getAnnotation(Symbol.class);
+                    if (s != null) {
+                        for (String t : s.value()) {
+                            if (t.equals(symbol)) {
+                                Descriptor d = (Descriptor) jenkins.getInjector().getInstance(e);
+                                if (type.isAssignableFrom(d.clazz)) {
+                                    cache.put(k, d);
+                                    return d;
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+
+            // not caching negative result since new plugins might be added later
+            return null;
+        } catch (IOException e) {
+            LOGGER.log(Level.WARNING, "Unable to find @Symbol",e);
+            return null;
+        }
+    }
+
     private static class Key {
+        private final String tag;
```

Comment:
This is a very local concern to one class, so it doesn't bother me. I won't do anything unless this is something you care a lot.